pipelines/batch/src/main/java/org/openmrs/analytics/FetchSearchPageFn.java

```diff
@@ -159,6 +160,10 @@ private static synchronized JdbcConnectionUtil getJdbcConnectionUtil(
   public void setup() throws SQLException, PropertyVetoException {
     log.info("Starting setup for stage " + stageIdentifier);
     fhirContext = FhirContext.forR4Cached();
+    // The documentation for `FhirContext` claims that it is thread-safe but looking at the code,
+    // it is not obvious if it is. This might be an issue when we write to it, like the next line.
+    fhirContext.setParserOptions(
+        new ParserOptions().setOverrideResourceIdWithBundleEntryFullUrl(false));
```

Comment:
we also define this on line 198 (I can't add a comment on that line on GitHub for some reason):

```
    parser.setOverrideResourceIdWithBundleEntryFullUrl(false);
```

do we need both, or can we get away with just having one call of this method?

Reply:
I don't think we need both; I removed that and moved part of its comment to here.
Just to clarify the reason we needed this here too is that the `parser` that is created below is not used when we fetch resources from a HAPI server (that parser is inside the HAPI client code). Added a comment for this too.